redaxo/src/addons/be_style/functions/rex_be_style_compile.php

```diff
@@ -0,0 +1,28 @@
+<?php
+/**
+ * Functions for be_style.
+ *
+ * @package redaxo\be_style
+ */
+
+/**
+ * Converts Backend SCSS files to CSS.
+ */
+function rex_be_style_compile()
+{
+    $beStyleAddon = rex_addon::get('be_style');
+
+    $compiler = new rex_scss_compiler();
+
+    $scss_files = rex_extension::registerPoint(new rex_extension_point('BE_STYLE_SCSS_FILES', [$beStyleAddon->getPath('scss/master.scss')]));
+    $compiler->setScssFile($scss_files);
+    //$compiler->setScssFile($this->getPath('scss/master.scss'));
+
+    // Compile in backend assets dir
+    $compiler->setCssFile($beStyleAddon->getPath('assets/css/styles.css'));
+
+    $compiler->compile();
+
+    // Compiled file to copy in frontend assets dir
+    rex_file::copy($beStyleAddon->getPath('assets/css/styles.css'), $beStyleAddon->getAssetsPath('css/styles.css'));
```

Comment:
Wäre es nicht möglich, hier einen ExtensionPoint zu notieren, der dann Objekte von PlugIns/andere AddOns entgegennimmt, die
- den Pfad zur master.scss (beinhalten die @import zu den anderen Scss Dateien) enthält
- den Pfad zur Datei wohin kompiliert werden soll
- den Pfad wohin die kompilierte Datei kopiert wird

Die Instanzen werden dann in Schleife abgearbeitet.

Reply:
er hat den code 1:1 übernommen wie vorher. bisher hat hier anscheinend noch niemand einen EP gebraucht.

denkst du es ist notwendig dass man hier dann auch nochmal die zieldatei anpassen will/soll? brauch man das? (es gibt ja bereits 10 zeilen drüber auch noch einen EP wg. dem scss).

Reply:
bisher gab es keinen. Wie Markus schon sagt, ist der Code 1:1 aus der boot.php extrahiert. 

Wenn ein EP notwendig seien sollte, würde ich das lieber von diesem PR trennen und in einem getrennten Issue/Pr abhandeln.

Reply:
Es ging doch darum, dass der Command nur das be_style kompiliert und andere PlugIns und AddOns nicht mit angestoßen werden können. Bisher musste sich jedes PlugIn/AddON selbst darum kümmern. Jetzt würde ich eine Möglichkeit schaffen, dass die Anderen sich registrieren können und kein weiteres Zutun von deren Seite aus notwendig wäre.